### PR TITLE
Prioritize @ autocomplete results based on recency and thread activity

### PR DESCRIPTION
**Pull request by nixusUM**
#### Summary
Prioritize @ autocomplete results based on recency and thread activity

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/21235

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: TECNO P8 Android, Samsung Galaxy S7 Edge Android, iPhone 11 Pro

#### Screenshots
image

#### Release Note
```release-note
Changed prioritize @ autocomplete results based on recency and thread activity
```

### CODE REVIEW

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (enahum) [inline app/components/autocomplete/at_mention/at_mention.tsx:243]: Why I left join? I expect that a DM channel has a `Channel` and `MyChannel` relation met. So this could be an inner join that should improve the query plan.

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (enahum) [inline app/components/autocomplete/at_mention/at_mention.tsx:310]: you could should make `sortedMembersId` a Set so that the filter is more performant

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (nixusUM) [reply app/components/autocomplete/at_mention/at_mention.tsx:151]: In the task description: remove the section headings for “Not in channel” and “Channel members”, and replace with a single heading called “Users”.

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (nixusUM) [reply app/components/autocomplete/at_mention/at_mention.tsx:173]: In the task description: remove the section headings for “Not in channel” and “Channel members”, and replace with a single heading called “Users”.

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (enahum) [reply app/components/autocomplete/at_mention/at_mention.tsx:11]: please update the import here

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (enahum) [inline app/components/autocomplete/at_mention/at_mention.tsx:12]:
```suggestion
import {logError} from '@utils/log';
```

please update the import here

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (enahum) [inline app/components/autocomplete/at_mention/at_mention.tsx:252]: this function should be moved to the `queries` folder

**Review by enahum — CHANGES_REQUESTED** [review]
make sure that after you change the imports that they are sorted correctly

File: app/components/autocomplete/at_mention/at_mention.tsx
Comment (larkox) [inline app/components/autocomplete/at_mention/at_mention.tsx:341]: This is doing nothing, isn't? Shouldn't we be using directly `users`?

File: assets/base/i18n/en.json
Comment (larkox) [inline assets/base/i18n/en.json:930]: These changes seem unrelated.

**Review by larkox — CHANGES_REQUESTED** [review]
Apart of what enahum already said, a few minor comments.

File: assets/base/i18n/en.json
Comment (enahum) [reply assets/base/i18n/en.json]: is fine, this is automatically generated.

File: app/queries/servers/user.ts
Comment (acrossoffwest) [inline app/queries/servers/user.ts:150]: do you logging errors?)

**Review by larkox — APPROVED** [review]

**Comment by nixusUM** [comment]
@enahum Hi, any update?

**Review by enahum — APPROVED** [review]

File: assets/base/i18n/en.json
Comment (enahum) [reply assets/base/i18n/en.json:1]: When you run i18n-extract it removes string that did not find and add those that were missed.

I'd say we should have kep the changes,

**Comment by enahum** [comment]
@nixusUM please fix the lint problems

**Comment by nixusUM** [comment]
@enahum Hi, I fixed lint problems

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@larkox what else is needed for us to merge this PR?

**Comment by larkox** [comment]
@enahum Nothing from my side I think. We are still missing @esethna review (I guess mostly to verify this is the expected behavior) and QA review. I will start the QA review process while we wait for the other review.

**Review by esethna — APPROVED** [review]
Exciting to see this get in, thanks @nixusUM! I'll defer to @yasserfaraazkhan for functionality check, please loop me in as needed if there's concerns.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**
